cli/tools/upgrade.rs

```diff
@@ -265,15 +265,13 @@ pub async fn upgrade(
 ) -> Result<(), AnyError> {
   let ps = ProcState::build(flags).await?;
   let current_exe_path = std::env::current_exe()?;
-  let output_exe_path =
-    upgrade_flags.output.as_ref().unwrap_or(&current_exe_path);
-  let metadata = fs::metadata(output_exe_path)?;
```

Comment:
Can we just change this to never throw?

Reply:
Agreed. The fix is easier than the revert.

Reply:
I'm not sure I understand how to change that... `metadata` and `permissions` by extension are used further down in the file. How should that be changed?

Reply:
At least a quick scroll down says that metadata is only used to check for permissions and root ownership. All of that is unnecessary if the file doesn't exist yet and thus could be handled in a `if let Some()` branch I think.

Reply:
In https://github.com/denoland/deno/pull/18207 I've updated the code to fall back to the current_exe_path metadata if there is none for the output_exe_path. Would that be sufficient to fix the issue? (it does fix the broken test)